src/resolve/resolved-dns-stub.c

```diff
@@ -751,6 +754,7 @@ static void dns_stub_query_complete(DnsQuery *query) {
                         _cleanup_(dns_packet_unrefp) DnsPacket *reply = NULL;
 
                         r = dns_stub_patch_bypass_reply_packet(&reply, q->answer_full_packet, q->request_packet,
+                                        DNS_PACKET_CD(q->question_bypass) && FLAGS_SET(q->flags, SD_RESOLVED_NO_VALIDATE),
```

Comment:
can you maybe prefix this with `/* checking_disabled= */` to make clear what this is

Reply:
we set both CD and AD only if DO is on in the question, no?